plugins/inputs/internet_speed/internet_speed.go

```diff
@@ -97,16 +100,30 @@ func (is *InternetSpeed) findClosestServer() error {
 		return fmt.Errorf("no servers found")
 	}
 
-	// return the first match
-	for _, server := range serverList {
+	// return the first match or the server with the lowest latency
+	// when filter mismatch all servers.
+	var min int64 = math.MaxInt64
+	selectIndex := -1
+	for index, server := range serverList {
 		if is.serverFilter.Match(server.ID) {
-			is.server = server
-			is.Log.Debugf("using server %s in %s (%s)\n", is.server.ID, is.server.Name, is.server.Host)
-			return nil
+			selectIndex = index
+			break
+		}
+		if server.Latency > 0 {
```

Comment:
I ran this PR locally, and never got any responses as nothing had latency reported, which then excludes all servers:

```
8182 in Boise, ID (boise1.cabospeed.com:8080): -1
10133 in Boise, ID (boise.speedtest.centurylink.net:8080): -1
1456 in Boise, ID (boi-speedtest.syringanetworks.net:8080): -1
14088 in Boise, ID (boi-speedtest.tekfinity.net.prod.hosts.ooklaserver.net:8080): -1
54619 in Boise, ID (speedtest02.z4internet.com:8080): -1
48001 in Mountain Home, ID (speedtest4.fybercom.net.prod.hosts.ooklaserver.net:8080): -1
4082 in La Grande, OR (speedtestnet1.eoni.com.prod.hosts.ooklaserver.net:8080): -1
28971 in Rupert, ID (slc.speedtest.safelink.net:8080): -1
20615 in Salmon, ID (speedtest3.fybercom.net.prod.hosts.ooklaserver.net:8080): -1
32266 in Albion, ID (speedtest-alb-2.atcnet.net.prod.hosts.ooklaserver.net:8080): -1
2023-02-21T15:32:00Z E! [inputs.internet_speed] Error in plugin: unable to find closest server: no server set: filter excluded all servers or no available server found
```

Reply:
-1 means the server timed out, please check if all servers are unavailable. You can download [binaries](https://github.com/showwin/speedtest-go/releases/tag/v1.4.1) here for checking.

Reply:
What are the odds that all 10 servers near by are down? :)

It seems that between v1.4.0 and v1.4.1 you now require root access to bind to a socket to run the test:

```
❯ ./speedtest-go --version
1.4.0
~/Downloads 
❯ ./speedtest-go -s 8182
Testing From IP: 159.118.149.74, (Sparklight) [43.6339, -116.2942] 
 
Target Server: [8182]     5.09km Boise, ID (United States) by Sparklight
Latency: 27.101596ms
Download Test: ......^C
❯ tar xvfz speedtest-go_1.4.1_Linux_x86_64.tar.gz 
LICENSE
README.md
speedtest-go
~/Downloads 
❯ ./speedtest-go --version
1.4.1
~/Downloads 
❯ ./speedtest-go -s 8182
Testing From IP: 159.118.149.74, (Sparklight) [43.6339, -116.2942] 

Target Server: [8182]     5.09km Boise, ID (United States) by Sparklight
2023/02/21 09:42:48 dial ip:icmp 72.24.184.146: socket: operation not permitted
```

Reply:
You are right, I will try to restore.

Reply:
Thank you, the change has been completed.

Reply:
Looks good, thanks! My only concern is I had one server get selected that was hundreds of miles away only because it had a low latency, but I think that could happen from time to time and folks could use an exclude if that became a problem.

Thanks again!

Reply:
@powersj Thanks, I think we can add a weight between latency and distance, and use this weight to judge the best server. Also the filter is still necessary which is a nice feature.

Reply:
In addition, if the list returned by the server contains far away servers, it means that the server may use a proxy or there are no servers nearby at all.

I think we should be more worried that the speed test result is not as fast as expected, because the current speedtest-go does not introduce auxiliary servers, I will add this feature in the next version. I think this might solve your concern.

Reply:
Awesome, thanks again for the PR!